### PR TITLE
[tests] Enable Code Coverage for JavaScript tests

### PR DESCRIPTION
**Pull request by apsdehal**
## Motivation and Context

Enables code coverage for JS files

## How Has This Been Tested

`npm run test_with_coverage` 

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by apsdehal** [comment]
@msbaines @mathfac Python lint job is failing due to unrelated unknown issue.

File: .circleci/config.yml [hunk 1]
```diff
@@ -278,30 +290,25 @@ jobs:
               #TODO: instead of reinstall, do this with configuration
               ./build.sh --headless
               pytest --cov-report=xml --cov=./ --cov-append tests/test_physics.py tests/test_sensors.py
+
+              # Generate JS CodeConv
+              npm run test_with_coverage
```
Comment (mathfac) [inline .circleci/config.yml]: Maybe, makes sense to have a separate step for JS testing?

File: .circleci/config.yml [hunk 2]
```diff
@@ -254,6 +254,18 @@ jobs:
...
+          name: Build Javascript bindings
+          command: |
+              # wait for npm and emscripten install
+              while [[ ! -f ~/emscripten_installed && ! -f ~/npm_deps_installed ]]; do sleep 2; done
+              cd habitat-sim
+              . ~/.bashrc
+              export PATH=$HOME/miniconda/bin:$PATH
+              . activate habitat
+              export EMSCRIPTEN=$HOME/emsdk/fastcomp/emscripten
+              nvm use v11.9.0
```
Comment (mathfac) [inline .circleci/config.yml]: For future that can be good to have versions mentioned somewhere in dependency or requirements specification and re-use it here.

Comment (apsdehal) [reply .circleci/config.yml]: Thanks. I will add to documentation in separate PR later.

File: .circleci/config.yml (hunk 1 above)
Comment (apsdehal) [reply .circleci/config.yml]: This is inside `run sim tests` so it should be good to keep all tests in same place as it runs CPP as well as Python tests? We do have a separate step also `js_lint_and_test` which tests without code coverage.

**Pull request merged**